core/src/common.cpp

```diff
@@ -407,14 +667,27 @@ namespace irods::http
 
 			// If we're running as a protected resource, assume we have a OIDC token
 			if (oidc_config_iter->at("mode").get_ref<const std::string&>() == "protected_resource") {
-				body_arguments args{{"token", bearer_token}, {"token_type_hint", "access_token"}};
+				nlohmann::json json_res;
+
+				// Use introspection endpoint if it exsists
+				if (auto introspection_endpoint_iter{irods::http::globals::oidc_endpoint_configuration().find(nlohmann::json::json_pointer{"/introspection_endpoint"})}; introspection_endpoint_iter != std::end(irods::http::globals::oidc_endpoint_configuration())) {
+					auto possible_json_res{validate_using_introspection_endpoint(bearer_token)};
+
+					if (!possible_json_res) {
+						return {.response = fail(status_type::unauthorized)};
+					}
+
+					json_res = *possible_json_res;
+				}
+				// Otherwise, try parsing as a JWT access token
+				else {
+					auto possible_json_res{validate_using_local_validation(bearer_token)};
 
-				auto json_res{hit_introspection_endpoint(url_encode_body(args))};
+					if (!possible_json_res) {
+						return {.response = fail(status_type::unauthorized)};
+					}
 
-				// Validate access token
-				if (!json_res.at("active").get<bool>()) {
-					logging::warn("{}: Access token is invalid or expired.", __func__);
-					return {.response = fail(status_type::unauthorized)};
+					json_res = *possible_json_res;
```

Comment:
We probably want to default to local validation. Either that or make order configurable?

Reply:
Changed to execute both validation methods (if possible), starting with local validation and failing to the introspection endpoint if it is available.

The current implementation means both methods are exhausted, even if a token is invalid (e.g., expired, garbage).